scapy/layers/dot15d4.py

```diff
@@ -336,9 +489,41 @@ class Dot15d4Cmd(Packet):
             9: "GTSReq"  # GTS request
             # 0x0a - 0xff reserved
         }),
-        # TODO command payload
+        # Secured Payload (variable length)
+        ConditionalField(
+            XStrField("sec_payload", ""),
+            lambda pkt: pkt.underlayer.getfieldval("fcf_security"),
+        ),
+        # Message Integrity Code (variable length)
+        ConditionalField(
+            XStrField("mic", ""),
+            lambda pkt: pkt.underlayer.getfieldval("fcf_security"),
+        ),
     ]
 
+    def post_dissect(self, s):
+        if self.underlayer.getfieldval("fcf_security"):
+            if self.aux_sec_header.sec_sc_seclevel in {1, 5}:
+                mic_length = 4
+            elif self.aux_sec_header.sec_sc_seclevel in {2, 6}:
+                mic_length = 8
+            elif self.aux_sec_header.sec_sc_seclevel in {3, 7}:
+                mic_length = 16
+            else:
+                mic_length = 0
```

Comment:
There seems to be a lot of code that is re-used. Could you exerternalize it to a function?